src/realizations/catchment/Bmi_C_Formulation.cpp

```diff
@@ -0,0 +1,161 @@
+    set_bmi_main_output_var(properties.at(BMI_REALIZATION_CFG_PARAM_REQ__MAIN_OUT_VAR).as_string());
+    set_forcing_file_path(properties.at(BMI_REALIZATION_CFG_PARAM_REQ__FORCING_FILE).as_string());
+    set_model_type_name(properties.at(BMI_REALIZATION_CFG_PARAM_REQ__MODEL_TYPE).as_string());
+    set_bmi_uses_forcing_file(properties.at(BMI_REALIZATION_CFG_PARAM_REQ__USES_FORCINGS).as_boolean());
+
+    // TODO: parse header fields and output variable names
+
+    // Then optional ...
+
+    if (properties.find(BMI_REALIZATION_CFG_PARAM_OPT__ALLOW_EXCEED_END) != properties.end()) {
+        set_allow_model_exceed_end_time(properties.at(BMI_REALIZATION_CFG_PARAM_OPT__ALLOW_EXCEED_END).as_boolean());
+    }
+
+    // Do this next, since we can construct the adapter and init the model as soon as we know whether it's present
+    auto other_in_var_it = properties.find(BMI_REALIZATION_CFG_PARAM_OPT__OTHER_IN_VARS);
+    // Construct the adapter, which also will handle the model initialization
+    if (other_in_var_it != properties.end()) {
+        // When "other_input_variables" was present, use the constructor that accepts those directly
+        set_bmi_model(std::make_shared<Bmi_C_Adapter>(
+                Bmi_C_Adapter(get_bmi_init_config(), get_forcing_file_path(), is_bmi_uses_forcing_file(),
+                              get_allow_model_exceed_end_time(), other_in_var_it->second, output)));
+    }
+    else {
+        set_bmi_model(std::make_shared<Bmi_C_Adapter>(
+                Bmi_C_Adapter(get_bmi_init_config(), get_forcing_file_path(), is_bmi_uses_forcing_file(),
+                              get_allow_model_exceed_end_time(), output)));
+    }
+
+    // Output variable subset and order, if present
+    auto out_var_it = properties.find(BMI_REALIZATION_CFG_PARAM_OPT__OUT_VARS);
+    if (out_var_it != properties.end()) {
+        std::vector<geojson::JSONProperty> out_vars_json_list = out_var_it->second.as_list();
+        vector_str out_vars(out_vars_json_list.size());
+        for (int i = 0; i < out_vars_json_list.size(); ++i) {
+            out_vars[i] = out_vars_json_list[i].as_string();
+        }
+        set_output_variable_names(out_vars);
+    }
+    // Otherwise, just take what literally is provided by the model
+    else {
+        set_output_variable_names(get_bmi_model()->GetOutputVarNames());
+    }
+
+    // Output header fields, if present
+    auto out_headers_it = properties.find(BMI_REALIZATION_CFG_PARAM_OPT__OUT_HEADER_FIELDS);
+    if (out_headers_it != properties.end()) {
+        std::vector<geojson::JSONProperty> out_headers_json_list = out_var_it->second.as_list();
+        vector_str out_headers(out_headers_json_list.size());
+        for (int i = 0; i < out_headers_json_list.size(); ++i) {
+            out_headers[i] = out_headers_json_list[i].as_string();
+        }
+        set_output_header_fields(out_headers);
+    }
+    else {
+        set_output_header_fields(get_output_variable_names());
+    }
```

Comment:
I could see a capability here (similar to a prior comment about overriding info in config file) where, like in other formulations, we specifiy params/info directly in the ngen config.  We can initialize the model, then use the "get_var" functions to verify a provided parameter is used by the model then the "set value" to set it to provided input.